dowhy/causal_refuters/add_unobserved_common_cause.py

```diff
@@ -39,22 +40,29 @@ def __init__(self, *args, **kwargs):
         minimum and maximum effect strength of observed confounders on treatment
         and outcome respectively.
 
+        :param simulation_method: The method to use for simulating effect of unobserved confounder. Possible values are ["direct-simulation", "linear-partial-r2", "non-parametric-partial-r2"].
         :param confounders_effect_on_treatment: str : The type of effect on the treatment due to the unobserved confounder. Possible values are ['binary_flip', 'linear']
         :param confounders_effect_on_outcome: str : The type of effect on the outcome due to the unobserved confounder. Possible values are ['binary_flip', 'linear']
-        :param effect_strength_on_treatment: float, numpy.ndarray: This refers to the strength of the confounder on treatment. For a linear effect, it behaves like the regression coeffecient. For a binary flip it is the probability with which it can invert the value of the treatment.
-        :param effect_strength_on_outcome: float, numpy.ndarray: This refers to the strength of the confounder on outcome. For a linear effect, it behaves like the regression coefficient. For a binary flip, it is the probability with which it can invert the value of the outcome.
-        :param effect_fraction_on_treatment: float: If effect_strength_on_treatment is not provided, this parameter decides the effect strength of the simulated confounder as a fraction of the effect strength of observed confounders on treatment. Defaults to 1.
-        :param effect_fraction_on_outcome: float: If effect_strength_on_outcome is not provided, this parameter decides the effect strength of the simulated confounder as a fraction of the effect strength of observed confounders on outcome. Defaults to 1.
+        :param effect_strength_on_treatment: float, numpy.ndarray: This refers to the strength of the confounder on treatment. Its interpretation depends on confounders_effect_on_treatment and the simulation_method. When simulation_method is direct-simulation, for a linear effect it behaves like the regression coefficient and for a binary flip, it is the probability with which effect of unobserved confounder can invert the value of the treatment. When simulation_method is linear-partial-r2 and non-parametric-partial-r2 (under a partial-linear DGP assumption), it refers to the partial R2 of the unobserved confounder wrt the treatment conditioned on the observed confounders. Only in the case of general non-parametric-partial-r2, it is (1-r), where r is the ratio of variance of reisz representer, alpha^2, based on observed confounders and that based on all confounders.
```

Comment:
This is a good point. For all the partial-r2 based methods (linear and non-parametric), it has the same interpretation. 
But the interpretation is different between direct_simulation and partial-r2 based simulations. 

So to avoid confusion, I have created two additional parameters. The first two (effect_strength) are used for direct_simulation, while the next two (partial_r2_confounder) are used for the partial-r2 based methods.

This is still a hacky solution. The ideal solution will be achieved in the new API where we can wrap the method specific parameters within the call to the method class. (and create separate method names and remove the simulation method parameter).

```
refute_estimate(..., method=AddSimulatedConfounder(effect_strength_on_treatment=a, effect_strength_on_outcome=b))
# if we want to use partial R2 methods
refute_estimate(..., method=NonParametricPartialR2Sensitivity(partial_r2_confounder_treatment=a, partial_r2_confounder_outcome=b))
```

Reply:
Thanks for making this change.  It is clearer, and will hopefully also make it easier to separate further in the future.